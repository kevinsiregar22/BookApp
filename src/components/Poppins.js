import React from 'react';
import {Text, StyleSheet} from 'react-native';

const Poppins = ({
  type = 'Regular',
  children,
  color = 'black',
  size = 14,
  textAlign = 'center',
}) => {
  //styled
  const style = StyleSheet.create({
    text: {
      fontFamily: `Poppins-${type}`,
      color: color,
      fontSize: size,
      textAlign: textAlign,
    },
  });
  return (
    <Text testID="text component" style={style.text}>
      {children}
    </Text>
  );
};

export default Poppins;
