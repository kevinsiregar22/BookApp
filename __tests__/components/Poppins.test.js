import React from 'react';
import Poppins from '../../src/components/Poppins';
import {render} from '@testing-library/react-native';

describe('Poppins Testing', () => {
  const {getByTestId, toJSON} = render(<Poppins color="white">audy</Poppins>);
  test('render with 14px size', () => {
    const element = getByTestId('text component');
    expect(element).toBeTruthy();
    expect(element.props.style.fontSize).toEqual(14);
    expect(element.props.style.color).toEqual('white');
    expect(toJSON()).toMatchSnapshot();
  });
});
