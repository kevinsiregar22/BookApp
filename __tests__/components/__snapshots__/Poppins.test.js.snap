// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poppins Testing render with 14px size 1`] = `
<Text
  style={
    Object {
      "color": "white",
      "fontFamily": "Poppins-Regular",
      "fontSize": 14,
      "textAlign": "center",
    }
  }
  testID="text component"
>
  audy
</Text>
`;
